languages/c/dockerfiles/c-23.Dockerfile

```
@@ -0,0 +1,40 @@
+# syntax=docker/dockerfile:1.7-labs
+FROM gcc:14.2.0-bookworm
+
+# Ensures the container is re-built if dependency files change
+ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="vcpkg.json,vcpkg-configuration.json"
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y zip=3.* && \ 
+    apt-get install --no-install-recommends -y g++=4:* && \
+    apt-get install --no-install-recommends -y build-essential=12.* && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# cmake is required by vcpkg
+RUN wget --progress=dot:giga https://github.com/Kitware/CMake/releases/download/v3.30.5/cmake-3.30.5-Linux-x86_64.tar.gz && \
+    tar -xzvf cmake-3.30.5-Linux-x86_64.tar.gz && \
+    mv cmake-3.30.5-linux-x86_64/ /cmake
+
+ENV CMAKE_BIN="/cmake/bin"
+ENV PATH="${CMAKE_BIN}:$PATH"
+
+RUN git clone https://github.com/microsoft/vcpkg.git && \
+    ./vcpkg/bootstrap-vcpkg.sh -disableMetrics
+
+ENV VCPKG_ROOT="/vcpkg"
+ENV PATH="${VCPKG_ROOT}:$PATH"
+
+WORKDIR /app
+
+# .git & README.md are unique per-repository. We ignore them on first copy to prevent cache misses
+COPY --exclude=.git --exclude=README.md . /app
+
+RUN vcpkg install --no-print-usage
+
+# Install language-specific dependencies
+RUN .codecrafters/compile.sh
+
+RUN mkdir -p /app-cached/build
```

```suggestion
RUN mkdir -p /app-cached
```

(`mv` should automatically handle creating the directory if I'm not wrong)